File: baseLib/src/test/java/me/ycdev/android/lib/common/type/IntegerHolderTest.java
package me.ycdev.android.lib.common.type;

import org.junit.Test;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

public class IntegerHolderTest {
    @Test
    public void basic() {
        {
            IntegerHolder holder = new IntegerHolder(0);
            assertThat(holder.value, is(0));
        }

        {
            IntegerHolder holder = new IntegerHolder(-10);
            assertThat(holder.value, is(-10));
        }

        {
            IntegerHolder holder = new IntegerHolder(100);
            assertThat(holder.value, is(100));
        }
    }
}

File: baseLib/src/test/java/me/ycdev/android/lib/common/type/IntegerHolderTest.kt
package me.ycdev.android.lib.common.type

import com.google.common.truth.Truth.assertThat
import org.junit.Test

class IntegerHolderTest {
    @Test
    fun basic() {
        run {
            val holder = IntegerHolder(0)
            assertThat(holder.value).isEqualTo(0)
        }

        run {
            val holder = IntegerHolder(-10)
            assertThat(holder.value).isEqualTo(-10)
        }

        run {
            val holder = IntegerHolder(100)
            assertThat(holder.value).isEqualTo(100)
        }
    }
}
